installJDOnSLURM.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python
+
+"""
+This script provides installation functions to launch a jupyter dask on SURF infrastructure (snellius or spider)
+with ssh access to the remote supercomputer administered by SLURM. This does not configure or set-up dCache.
+This script is automatically invoked from the runJupyterDaskOnSLURM script when install or uninstall flags are invoked.
+
+USE:
+The script can most easily be invoked through 'python runJupyterDaskOnSLURM.py --<your arg>'
+
+Paths to the working directory and scripts on the remote host, as well as to the local connection configuration file can be specified below.
+Configurable paths are:
+
+config_path     : path to config file that is created/updated. default is in local installation of JupyterDaskOnSLURM
+                  repository. Can be adapted to user preferences
+remoteWD        : Working directory from which to submit batch job on remote host
+remoteJDD       : Clone directory of JupyterDaskOnSLURM
+remoteScriptWD  : Path to directory on remote host where job submission scripts are located
+                  ATTENTION! When adding a job script on a plattform other than spider/snellius @SURF the user MUST
+                  create a job script for the platform (this can be done by using the existing scripts as templates).
+                  Expected naming convention is 'jupyter_dask_xxx.bsh', where xxx is the name of the platform.
+mamba_URL       : Download URL for mamba installation file
+"""
+ 
+from fabric import Connection
+from runJupyterDaskOnSLURM import ssh_remote_executor
+
+config_path = './config/platforms/platforms.ini'
+remoteWD = '~'
+remoteJDD = '~/JupyterDaskOnSLURM'
+mamba_URL = 'https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-x86_64.sh'
+
+def check_clone(conn):
+    """
+    Check if repository has been cloned already on remote host
+
+    :param conn: ssh connection object
+    :return folder_exists: Logical on whether the repository exists on remote host 
+    """
+    cmd = f"if test -d JupyterDaskOnSLURM; then echo 'True'; fi"
+    folder_exists = False
+    result = conn.run(cmd, hide=True)
+    folder_exists = result.stdout
+    return folder_exists
+
+def clone_folder(conn):
+    """
+    Clone repository from github to remote host
+
+    :param conn: ssh connection object
+    :return None:
+    """
+    cmd = "git clone https://github.com/RS-DAT/JupyterDaskOnSLURM.git"
+    conn.run(cmd, hide=True)
+    return None
+
+def test_mamba(conn):
+    """
+    Check if mamba is installed on remote host
+
+    :param conn: ssh connection object
+    :return mamba_exists: Logical on if mamba is installed on remote host
+    """
+    cmd = "which mamba"
+    result = conn.run(cmd, hide=True)
```

Comment:
running from mac on pristine SPIDER surf account causes fatal exit

`
File "/Users/eslt0101/mambaforge/envs/RS-DAT/lib/python3.9/site-packages/invoke/runners.py", line 509, in _finish
    raise UnexpectedExit(result)
invoke.exceptions.UnexpectedExit: Encountered a bad command exit code!

Command: 'which mamba'

Exit code: 1

Stdout:



Stderr:

/usr/bin/which: no mamba in (/home/stursdat-mgrootes/.local/bin:/home/stursdat-mgrootes/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin)
`
it seems that invoke.run (which is wrapped by connection.run) doesn't  pass exceptions out but raises them. Did you test this? Is this only me?

Reply:
This works on windows WSL without raising a fatal exit. Maybe there is a better way to check if mamba is installed. Any thoughts on how we can?